src/pip/_internal/vcs/versioncontrol.py

```diff
@@ -347,24 +351,24 @@ def get_url_rev_and_auth(cls, url):
 
     @staticmethod
     def make_rev_args(username, password):
-        # type: (Optional[str], Optional[str]) -> List[str]
+        # type: (Optional[str], Optional[str]) -> CommandArgs
         """
         Return the RevOptions "extra arguments" to use in obtain().
         """
         return []
 
     def get_url_rev_options(self, url):
-        # type: (str) -> Tuple[str, RevOptions]
+        # type: (HiddenText) -> Tuple[HiddenText, RevOptions]
         """
         Return the URL and RevOptions object to use in obtain() and in
         some cases export(), as a tuple (url, rev_options).
         """
-        url, rev, user_pass = self.get_url_rev_and_auth(url)
+        secret_url, rev, user_pass = self.get_url_rev_and_auth(url.secret)
         username, password = user_pass
         extra_args = self.make_rev_args(username, password)
```

Comment:
We could do `hide_value(password)`, and make the accompanying type changes for `self.make_rev_args` and `RevOptions`.